tests/system/action/poll/test_vote.py

```diff
@@ -0,0 +1,382 @@
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [{"id": 1, "user_id": 1, "value": {"11": 1}}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        vote = self.get_model("vote/1")
+        assert vote.get("value") == "Y"
+        assert vote.get("option_id") == 11
+        assert vote.get("weight") == "1.000000"
+        assert vote.get("meeting_id") == 113
+        assert vote.get("user_id") == 1
+        option = self.get_model("option/11")
+        assert option.get("vote_ids") == [1]
+        assert option.get("yes") == "1.000000"
+        assert option.get("no") == "0.000000"
+        assert option.get("abstain") == "0.000000"
+        user = self.get_model("user/1")
+        assert user.get("vote_$_ids") == ["113"]
+        assert user.get("vote_$113_ids") == [1]
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [{"id": 1, "user_id": 2, "value": {"11": 0}}],
+                }
+            ],
+        )
+        self.assert_status_code(response, 200)
+        vote = self.get_model("vote/2")
+        assert vote.get("value") == "Y"
+        assert vote.get("option_id") == 11
+        assert vote.get("weight") == "0.000000"
+        assert vote.get("meeting_id") == 113
+        assert vote.get("user_id") == 2
+        option = self.get_model("option/11")
+        assert option.get("vote_ids") == [1, 2]
+        assert option.get("yes") == "1.000000"
+        assert option.get("no") == "0.000000"
+        assert option.get("abstain") == "0.000000"
+        user = self.get_model("user/2")
+        assert user.get("vote_$_ids") == ["113"]
+        assert user.get("vote_$113_ids") == [2]
+
+    def test_vote_correct_pollmethod_YN(self) -> None:
+        self.create_model("group/1", {"user_ids": [1]})
+        self.create_model("option/11", {"meeting_id": 113, "poll_id": 1})
+        self.create_model("option/12", {"meeting_id": 113, "poll_id": 1})
+        self.create_model("option/13", {"meeting_id": 113, "poll_id": 1})
+        self.create_model(
+            "poll/1",
+            {
+                "title": "my test poll",
+                "option_ids": [11, 12, 13],
+                "pollmethod": "YN",
+                "meeting_id": 113,
+                "entitled_group_ids": [1],
+            },
+        )
+        self.create_model("meeting/113", {"name": "my meeting"})
+        self.update_model("user/1", {"is_present_in_meeting_ids": [113]})
+        response = self.client.post(
+            "/",
+            json=[
+                {
+                    "action": "poll.vote",
+                    "data": [
+                        {
+                            "id": 1,
+                            "user_id": 1,
+                            "value": {"11": "Y", "12": "N", "13": "A"},
```

Comment:
This should be invalid: In a `YN` poll, you cannot give `A` as a value.

Reply:
Add error raising for this.